### ISSUE TITLE
Ternary expression resolves to strange supertype

### ISSUE THREAD

**Issue by gabor-farkas** [janino-compiler/janino#85]
Hello,

Evaluating:
```java
List list = true ? new ArrayList() : Arrays.asList(new String [] {});
```
Doesn't compile, with the error message: _Assignment conversion not possible from type "java.util.Collection" to type "java.util.List"_

I prepared this in the form of a unit test: https://github.com/gabor-farkas/janino-ternary-expression
It works with 3.0.11 but fails with 3.0.12

**Comment by aunkrig** [comment]
Thank you for your very precise bug report!

Reproduced, diagnosed and fixed.

The bug was introduced in the context of

> The <code>?:</code> operator no longer issues an error when the MHS is not assignable from the RHS and vice versa; instead, the expression type is the "common base type" of the MHS and RHS.

Please re-test!

**Comment by gabor-farkas** [comment]
Thanks for the quick fix!
I tested and I can confirm it works fine now!

**Comment by aunkrig** [comment]
Great! Do you want me to create a new Janino release?

**Comment by gabor-farkas** [comment]
If it's not planned otherwise to happen soon, then yes, it would be helpful, thanks!

**Comment by aunkrig** [comment]
FYI: I just released version 3.0.13, so the fix is now available in an official release.

**Comment by gabor-farkas** [comment]
Thanks! I'll give it a try.
